### PR TITLE
Modified plasma_dispersion_func to use wofz() in scipy. Fixed tests

### LINKED ISSUE #208: Use Dawson function for dispersion
`plasma_dispersion_func` under `mathematics.py` currently uses `erf()` along with some other terms. This can be simplified to Dawson function, [dawsn](https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.dawsn.html), and may even offer some minor speedups if scipy implements it in C code.

### PR DESCRIPTION
**Pull request by siddharth185**
Modified plasma_dispersion_func to use wofz() in scipy as Faddeeva function in mathematic.py . Modified rtol values so the tests pass. Modified the values in Docstring so the Doctests pass.

Closes #208

### CODE REVIEW

File: plasmapy/mathematics/mathematics.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import numpy as np
 from scipy import special
 from astropy import units as u
+from scipy.special import wofz as Faddeeva_function
```
Comment (StanczakDominik) [inline plasmapy/mathematics/mathematics.py]: ... huh, that was there all along? :laughing:

File: plasmapy/mathematics/tests/test_dispersion.py [hunk 2]
```diff
@@ -35,7 +35,7 @@ def test_plasma_dispersion_func(w, expected):
 
     Z_of_w = plasma_dispersion_func(w)
 
-    assert np.isclose(Z_of_w, expected, atol=1e-12 * (1 + 1j), rtol=1e-12), \
+    assert np.isclose(Z_of_w, expected, atol=1e-12 * (1 + 1j), rtol=1e-6), \
```
Comment (StanczakDominik) [inline plasmapy/mathematics/tests/test_dispersion.py]: I was pretty much guessing blindly with the 1e-6 value. Could you try something lower, closer to the old 1e-12? Go as low as you can - ideally we'd like to both be accurate in testing the old tabularized values and not so accurate that machine precision will break these tests.

**Comment by StanczakDominik** [comment]
As for the test failures... suddenly all the quantity docstrings seem to be failing, but @siddharth185 didn't change anything related to that. I checked on master at my own fork and well hello there, they're failing on master. This is looking like an @astropy change... I'm looking into it, because I think a potential change on their end may have fixed #135.

File: plasmapy/mathematics/tests/test_dispersion.py (hunk 2 above)
Comment (siddharth185) [reply plasmapy/mathematics/tests/test_dispersion.py]: Okay! I'll work on this.

**Comment by StanczakDominik** [comment]
The newest changes on master should solve the docstring problem.

**Comment by StanczakDominik** [comment]
I'm pretty sure this can be merged after it's rebased on top of master! I'll make sure and merge it tonight.

**Comment by StanczakDominik** [comment]
Merging it. Thanks, @siddharth185 !

**Pull request merged**